src/OrleansRuntime/Storage/ProviderManagerBase.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Orleans.Providers;
+using Orleans.Runtime.Configuration;
+using Orleans.Storage;
+
+namespace Orleans.Runtime.Storage
+{
+    internal abstract class ProviderManagerBase<TProvider> : IProviderManagerBase<TProvider> where TProvider : IProvider
+    {
+        protected readonly IProviderRuntime providerRuntime;
+        protected ProviderLoader<TProvider> providerLoader;
+
+        public ProviderManagerBase(IGrainFactory grainFactory, IServiceProvider serviceProvider, IProviderRuntime providerRuntime)
+        {
+            this.providerRuntime = providerRuntime;
+            GrainFactory = grainFactory;
+            ServiceProvider = serviceProvider;
+        }
+
+        public abstract string ProviderCategory { get; }
+
+        internal Task LoadProviders(IDictionary<string, ProviderCategoryConfiguration> configs)
+        {
+            providerLoader = new ProviderLoader<TProvider>();
+
+            if (!configs.ContainsKey(ProviderCategory))
+                return TaskDone.Done;
+
+            providerLoader.LoadProviders(configs[ProviderCategory].Providers, this);
+            return providerLoader.InitProviders(providerRuntime);
+        }
+
+        public Task CloseProviders()
+        {
+            List<Task> tasks = new List<Task>();
+            foreach (var provider in GetProviders())
+            {
+                tasks.Add(provider.Close());
+            }
+            return Task.WhenAll(tasks);
+        }
+
+        public int GetNumLoadedProviders()
+        {
+            return providerLoader.GetNumLoadedProviders();
+        }
+
+        public IList<TProvider> GetProviders()
+        {
+            return providerLoader.GetProviders();
+        }
+
+        public Logger GetLogger(string loggerName)
+        {
+            return LogManager.GetLogger(loggerName, LoggerType.Provider);
+        }
+
+        public Guid ServiceId => providerRuntime.ServiceId;
+
+        public string SiloIdentity => providerRuntime.SiloIdentity;
+
+        public IGrainFactory GrainFactory { get; private set; }
+        public IServiceProvider ServiceProvider { get; private set; }
+        public void SetInvokeInterceptor(InvokeInterceptor interceptor)
+        {
+            providerRuntime.SetInvokeInterceptor(interceptor);
+        }
+
+        public InvokeInterceptor GetInvokeInterceptor()
+        {
+            return providerRuntime.GetInvokeInterceptor();
+        }
```

Comment:
seems odd to have all of these forwarder methods. Why is it not exposing provider runtime directly?